Source/inv.cpp

```diff
@@ -294,73 +294,67 @@ bool AutoEquip(int playerId, const Item &item, inv_body_loc bodyLocation, bool p
 
 void CheckInvPaste(Player &player, Point cursorPosition)
 {
-	int i = cursorPosition.x;
-	int j = cursorPosition.y;
-
 	Size itemSize = GetInventorySize(player.HoldItem);
 
-	if (!IsHardwareCursor()) {
-		// offset the cursor position to match the hot pixel we'd use for a hardware cursor
-		i += itemSize.width * INV_SLOT_HALF_SIZE_PX;
-		j += itemSize.height * INV_SLOT_HALF_SIZE_PX;
-	}
+	auto possibleSlot = [&]() -> std::optional<int> {

```

Comment:
I'm neither a fan of arrow functions or optional, though they can have there place.

Reply:
We could get rid of both by introducing something like `SLOTXY_INVALID = 255`, but I'd rather not do that.

Reply:
This could be a named function like `size_t FindSlotUnderCursor(Point hotPixelPosition)`. Return `SLOTXY_NOT_FOUND=-1` (or even `SLOTXY_END=NUM_XY_SLOTS`) and it matches the behaviour of other find algorithms like `string::find_first_of` or `std::find(it, end, criteria)`

Reply:
`NUM_XY_SLOTS` seems more sensible.

Done.